pages/create/project.vue

```diff
@@ -567,39 +795,85 @@
           <div class="image">
             <img
               :src="
-                item.preview
-                  ? item.preview
+                gallery[currentGalleryIndex].preview
+                  ? gallery[currentGalleryIndex].preview
                   : 'https://cdn.modrinth.com/placeholder-banner.svg'
               "
               alt="preview-image"
             />
             <div class="bottom">
               <SmartFileInput
+                max-size="5242880"
                 accept="image/png,image/jpeg,image/gif,image/webp,.png,.jpeg,.gif,.webp"
                 prompt="Upload"
-                @change="(files) => showGalleryPreviewImage(files, index)"
+                :class="{
+                  'known-error':
+                    !gallery[currentGalleryIndex].preview &&
+                    showKnownGalleryErrors,
+                }"
+                @change="
+                  (files) => showGalleryPreviewImage(files, currentGalleryIndex)
+                "
               />
+              <div>
+                <button
+                  class="iconified-button"
+                  @click="
+                    gallery[currentGalleryIndex].icon = null
+                    gallery[currentGalleryIndex].preview = null
+                    $forceUpdate()
+                  "
+                >
+                  <TrashIcon />
+                  Reset
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div class="gallery-items">
+          <div
+            v-for="(item, index) in gallery.filter((it) =>
+              currentGalleryIndex !== -1
```

Comment:
nit: ternaries with boolean literals can usually be more clearly written as boolean expressions:

```suggestion
currentGalleryIndex === -1 || it !== gallery[currentGalleryIndex]
```

(If I'm too nit-picky here btw, feel free to ignore and/or tell me off!)